src/core_plugins/kibana/public/visualize/editor/agg_params.js

```diff
@@ -82,11 +82,23 @@ uiModules
         // build collection of agg params html
         type.params.forEach(function (param, i) {
           let aggParam;
+          let fields;
+          // if field param exists, compute allowed fields
+          if (param.name === 'field') {
+            fields = $aggParamEditorsScope.indexedFields;
+            $aggParamEditorsScope.showAnalyzedWarning = true;
+            if (!_.isUndefined(param.showAnalyzedWarning) && !param.showAnalyzedWarning) {
+              // if not explicitly set, always show the warning about analyzed field
+              $aggParamEditorsScope.showAnalyzedWarning = false;
```

Comment:
It looks like `aggParam` is available in field.html's scope, so you should be able to do `aggParam.showAnalyzedWarning` instead of attaching this variable to the editor scope.